fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -113,9 +114,18 @@ public <M extends Message> Object eval(@Nonnull final FDBRecordStoreBase<M> stor
         final var fields = Objects.requireNonNull(getResultType().getFields());
         var i = 0;
         for (final var child : getChildren()) {
-            final var childResultElement = child.eval(store, context);
+            var childResultElement = child.eval(store, context);
             if (childResultElement != null) {
-                resultMessageBuilder.setField(descriptorForType.findFieldByNumber(fields.get(i).getFieldIndex()), childResultElement);
+                final var field = fields.get(i);
```

Comment:
Is there an actual test for this in record-layer?

Reply:
No I don't think so, do you know which test class I can add unit test for this?

Reply:
resolved offline